pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -525,8 +526,12 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
       routeProvider = _implicitHybridTableRouteProvider;
     }
 
+    // Get the appropriate routing manager based on query options
+    RoutingManager routingManager = _multiClusterRoutingContext != null && logicalTableConfig != null
+        ? _multiClusterRoutingContext.getRoutingManager(pinotQuery.getQueryOptions())
+        : _routingManager;
     // Get the tables hit by the request
-    TableRouteInfo routeInfo = routeProvider.getTableRouteInfo(tableName, _tableCache, _routingManager);
+    TableRouteInfo routeInfo = routeProvider.getTableRouteInfo(tableName, _tableCache, routingManager);
```

Comment:
It seems it's pretty easy to create a bug in a future code change where instead of passing `routingManager` someone uses `_routingManager`.

Maybe rename `_routingManager` to `_defaultRoutingManager` and add a comment on the var declaration?